go/private/rules/cgo.bzl

```diff
@@ -104,7 +111,7 @@ def cgo_configure(go, srcs, cdeps, cppopts, copts, cxxopts, clinkopts):
         if CcInfo in d:
             cc_transitive_headers = d[CcInfo].compilation_context.headers
             inputs_transitive.append(cc_transitive_headers)
-            cc_libs = _cc_libs(d)
+            cc_libs = _cc_libs_for_target(d)
```

Comment:
This list is now distinct from the libraries returned by `_cc_libs`, but the names are really similar. What's the distinction? Won't there be an error if there are differences?

Reply:
The old `_cc_libs` fished out the archive/dso path from each `LibraryToLink` in `d`, giving back a list of those paths. At that point, we no longer have `LibraryToLink.alwayslink` to inspect.

So, as the code stands now:

* `_cc_libs(library_to_link)`:  gets the archive/dso path for the given library (currently as singleton list, but as you pointed out, this ought to just return the path itself).
* `_cc_libs_for_target(target)`: returns the list of all lib paths for a given target (as `_cc_libs` _used to_).
* `_cc_libraries_to_link(target)`: returns the list of all `LibraryToLink`s for a given `target`.

The names are terrible, I will admit. Ultimately, my changes were intended to address the fact I need the respective `LibraryToLink` so I can inspect `alwayslink`.

With that said, I noticed something odd about the code before I got involved (my notes as comments):

```python
            
            # cc_libs has the path of all libs for target `d`
            cc_libs = _cc_libs(d)
            # the existing implementation immediately extends inputs_direct and deps_direct with cc_libs
            inputs_direct.extend(cc_libs)
            deps_direct.extend(cc_libs)
            # [... elided ...]
            for lib in cc_libs:
                if (lib.basename.startswith("lib") and
                    libname = lib.basename[len("lib"):lib.basename.rindex(".")]
                    clinkopts.extend(["-L", lib.dirname, "-l", libname])
                    inputs_direct.append(lib) # <===== isn't this superfluous, since we extended inputs_direct with all of cc_libs above???
                elif (lib.basename.startswith("lib") and
                      has_versioned_shared_lib_extension(lib.basename)):
                    libname = ":%s" % lib.basename
                    clinkopts.extend(["-L", lib.dirname, "-l", libname])
                    inputs_direct.append(lib) # <===== ditto
                else:
                    lib_opts.append(lib.path)
```

Assuming I'm tracking how the code is intended to work, my next commit (which I will push shortly) will greatly clean things up.